### ISSUE TITLE
Bots can get stuck at corners and ledges

### ISSUE THREAD

**Issue by sweet235** [Unvanquished/Unvanquished#2354]
Sometimes, a bot seems to be ignorant of map geometry, and gets stuck.

This can happen at ledges on the walls, like here (map-zittrig-arena):

![abot](https://user-images.githubusercontent.com/110394607/210513907-9837d671-ea89-4afc-8be4-ff0aa618ca1d.png)

It can also happen when the walls have weird shapes, like here (map-bbs):

![hbot](https://user-images.githubusercontent.com/110394607/210514019-c6de064c-7a42-4c86-8b51-3509b3389b29.png)

 When activated, the unstick behavior can fix most of these problems. But the bots tend to get killed before that.

**Comment by sweet235** [comment]
They can also get stuck at friendly buildables. But a jump will be triggered a few seconds later. But still, this is rather annoying to observe.

![a](https://user-images.githubusercontent.com/110394607/211141423-7cdcaa34-173a-4959-b5c1-7b9655ac422a.jpg)

**Comment by necessarily-equal** [comment]
I believe this the first part of this issue is a duplicate of https://github.com/Unvanquished/Unvanquished/issues/2263, at least when bots are not fighting.

I think the buildable issue is a different one. https://github.com/Unvanquished/Unvanquished/pull/2334 was supposed to fix a very similar issue, https://github.com/Unvanquished/Unvanquished/issues/2275. Do you have this fix on the affected server? The basic description seems very similar: bot stuck over buildables. Do you think this should be a different issue?

**Comment by cu-kai** [comment]
This is not fixed as of 0.54.0 and with the new ingame navmesh generator. I see rants getting stuck and trembling on the exact same ramp reported by @sweet235 

Amongst other bugs with the new navmesh generator.

**Comment by slipher** [comment]
The "new" navmesh generator is not intended to fix any bugs. It's just the same generator but running inside the gamelogic process.

**Comment by necessarily-equal** [comment]
The navmesh corner navigation is reputed fixed by https://github.com/Unvanquished/Unvanquished/pull/2396. I'm not sure about the buildable problem, I don't think I managed to reproduce it.

Maybe we should close this issue for now and open a new one should the buildable issue still trigger? (It seems a bit off-topic in this issue unfortunately)

**Comment by bmorel** [comment]
I think you should wait for the persons reporting the bug to confirm it no longer happens, instead of closing something you are not certain to have fixed.

Out-topic, but should add some information about the navmesh generator problems & benefits:

As for the "new" generator, it's port inside gamelogic indeed was not meant to fix any bug nor to bring any functionality.
Or rather, this brings the functionality to be able to generate navmeshes to gameplay changes, and AFAIK removes the functionality to be able to generate navmeshes in parallel (making it slower compared to calling daemonmap).
This still led to various regressions in the navmesh generator itself though (the patch geometry things not being ported for the most obvious example) and in how the game handles that (those I could identify a few which were fixed I believe). All is not bad though, as this also allows server owners to distribute smaller map files (which is something I've been using on my servers for a while).

**Comment by necessarily-equal** [comment]
> I think you should wait for the persons reporting the bug to confirm it no longer happens, instead of closing something you are not certain to have fixed.

(Well I didn't close this issue?)
